ext/crates/fp/src/matrix/subquotient.rs

```diff
@@ -184,6 +184,58 @@ impl Subquotient {
     }
 }
 
+#[cfg(feature = "proptest")]
+pub mod arbitrary {
+    use proptest::prelude::*;
+
+    use super::*;
+    use crate::matrix::subspace::arbitrary::SubspaceArbParams;
+    pub use crate::matrix::subspace::arbitrary::MAX_DIM;
+
+    #[derive(Debug, Clone)]
+    pub struct SubquotientArbParams {
+        pub p: Option<ValidPrime>,
+        pub dim: BoxedStrategy<usize>,
+    }
+
+    impl Default for SubquotientArbParams {
+        fn default() -> Self {
+            Self {
+                p: None,
+                dim: (0..=MAX_DIM).boxed(),
+            }
+        }
+    }
+
+    impl Arbitrary for Subquotient {
+        type Parameters = SubquotientArbParams;
+        type Strategy = BoxedStrategy<Self>;
+
+        fn arbitrary_with(args: Self::Parameters) -> Self::Strategy {
+            let p = match args.p {
+                Some(p) => Just(p).boxed(),
+                None => any::<ValidPrime>().boxed(),
+            };
+
+            (p, args.dim)
+                .prop_flat_map(|(p, dim)| {
+                    let sub = Subspace::arbitrary_with(SubspaceArbParams {
+                        p: Some(p),
+                        dim: Just(dim).boxed(),
+                    });
+                    let quotient = Subspace::arbitrary_with(SubspaceArbParams {
+                        p: Some(p),
+                        dim: Just(dim).boxed(),
+                    });
```

Comment:
Doesn't the quotient have to be a sub of the sub here?

Reply:
It would be helpful to update this file with a comment explaining the invariants that sub and quotient passed in are supposed to satisfy and what additional invariants are satisfied when in normal form.

Reply:
I agree it's weird that quotient doesn't have to define a subspace of sub, but that's because a `Subquotient` containing `Subspace`s `gens` and `quotient` really represents the space (`gens` + `quotient`) / `quotient`, mathematically speaking.

This is documented at `Subquotient::from_parts`, but maybe it should be somewhere else too?

Reply:
How about at the top of the file?

Reply:
Good idea, I'll move it there and add a link to it at `Subquotient::from_parts`